store/src/java/com/zimbra/cs/service/account/Auth.java

```diff
@@ -508,4 +528,36 @@ public static void addAccountToLogContextByAuthToken(Provisioning prov, AuthToke
         if (aid != null && !aid.equals(id))
             AccountUtil.addAccountToLogContext(prov, aid, ZimbraLog.C_ANAME, ZimbraLog.C_AID, null);
     }
+
+    public static AuthToken validateAndCreateNewJwtToken(String jwt, Account acct, Provisioning prov) throws ServiceException {
+        if (StringUtil.isNullOrEmpty(jwt)) {
+            throw ServiceException.AUTH_REQUIRED();
+        }
+        String tokenKey = acct.getJWTKey();
+        if (StringUtil.isNullOrEmpty(tokenKey)) {
+            throw ServiceException.AUTH_REQUIRED();
+        }
+        java.security.Key key = new SecretKeySpec(tokenKey.getBytes(StandardCharsets.UTF_8), SignatureAlgorithm.HS512.getJcaName());
+        Claims claims = null;
+        try {
+            claims = Jwts.parser().setSigningKey(key).parseClaimsJws(jwt).getBody();
+        } catch(ExpiredJwtException eje) {
+            throw ServiceException.AUTH_EXPIRED(eje.getMessage());
+        } catch(SignatureException se) {
```

Comment:
I think separate exception should be  thrown for SignatureException which indicates that signature is not valid.